pandas/io/parsers/base_parser.py

```diff
@@ -789,11 +789,16 @@ def _cast_types(self, values, cast_type, column):
 
         else:
             try:
-                values = astype_nansafe(values, cast_type, copy=True, skipna=True)
+                casted = astype_nansafe(values, cast_type, copy=True, skipna=True)
             except ValueError as err:
                 raise ValueError(
                     f"Unable to convert column {column} to type {cast_type}"
                 ) from err
+            if is_integer_dtype(cast_type) and not (casted == values).all():
```

Comment:
there might be overlap with something in dtypes.cast for this check

Reply:
Thanks, there is indeed! I changed this to use `maybe_cast_to_integer_array` for integers. Some of the overflows are only raised as a FutureWarning at this point, but will eventually be raised when the deprecation from #41734 becomes enforced. I tried to reflect this in the [test](https://github.com/pandas-dev/pandas/blob/88d86508d83907bafd4a44af26ecb8dfbe1ee469/pandas/tests/io/parser/common/test_ints.py#L146-L160) in a way that makes it simple to change to expected behavior later (even though it looks a bit ugly now).